### DISCUSSION TITLE
Internet access to render plantuml

### DISCUSSION THREAD

**Discussion by m-fonseca** [iamcco/markdown-preview.nvim#318] [Q&A]
I noticed when using plantuml code blocks that internet access was needed to render properly.  This has privacy concerns for me and was wondering if the code blocks were being sent to a remote location to be rendered, or the plantuml code was being fetched from a remote server and rendered locally in the browser ?

Scares me a little and I had to double check that the regular markdown didn't require internet access to render. 

Love the plug in.  Thank you.

**Comment by ThomasFeher** [comment]
I stumbled upon this as well, and I'm also concerned about that. As far as I
understand it, the code is send to the public PlantUML server which renders the
picture, the picture is retrieved from the server and embedded into the HTML
that is displayed by your browser.

This behaviour should be documented somewhere prominently, IMHO.

To work around that you need to start your own PlantUML server, which is quite
easy, see the PlantUML documentation for that. Then you need to configure that
server's location in your vimrc like that:

```
let g:mkdp_preview_options = {
    \ 'uml': {
    \     'server': 'http://your_plantuml_server_url:8080'
    \     }
    \ }
```

**Reply by m-fonseca** [reply]
Excellent.  At least there is a reasonable work around.  Just wish it was obvious.   Hmm I wonder if mermaid is like this too.  Need to go check.
